src/relations/mysql_provider.py

```diff
@@ -396,8 +303,12 @@ def _on_database_broken(self, event: RelationBrokenEvent) -> None:
             # run once by the leader
             return
 
+        if len(self.model.relations[DB_RELATION_NAME]) == 1:
+            # remove kubernetes service when last relation is removed
+            self.k8s_helpers.delete_endpoint_services(["primary", "replicas"])
```

Comment:
thoughts of tying the endpoint services with the charm and not the `mysql_client` relation? this way, we can [do what Mykola did](https://github.com/canonical/mysql-router-k8s-operator/blob/main/src/charm.py#L93-L131) to avoid manually deleting services

I have to test yet, but this may also have the benefit of not needing the `--trust` flag when deploying the charm

Reply:
Good call, that's a smart move.

Not sure about not having to trust though, will test it.